### PR TITLE
Create immutable copy to avoid concurrent modification exception

### PR DESCRIPTION
**Pull request by DanVanAtta**
Not 100% sure we'll avoid the below error:

```
! java.util.ConcurrentModificationException: null
```

> https://travis-ci.org/triplea-game/triplea/jobs/630844224?utm_medium=notification&utm_source=github_status

It does not seem we allow or could have a null collection element. Creating a copy of the passed in collection is still a good change as we do not want to rely on implementation that is passed to this method. If the method can be called in a concurrent situation, then it needs to try and deal with that (eg: hardened-API, it should not be up to a caller to make sure correct conditions exist before calling an implementation)

Rest of the stack trace:
```
! at org.triplea.server.http.web.socket.MessageBroadcaster.accept(MessageBroadcaster.java:26)
! at org.triplea.server.http.web.socket.MessageBroadcaster.accept(MessageBroadcaster.java:13)
! at org.triplea.server.lobby.game.listing.GameListingEventQueue.gameRemoved(GameListingEventQueue.java:42)
! at org.triplea.server.lobby.game.listing.GameListing.removeGame(GameListing.java:88)
```

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix:  
[] Other:   

## Testing

[] Manual testing done

### CODE REVIEW

**Comment by DanVanAtta** [comment]
That is some good sleuthing @RoiEXLab 

> Basically while the server is broadcasting that a new game has connected to all nodes, some game is removed from another thread which causes the exception

I think that is likely to be exactly right. Though, I added the sleep as you mentioned and the tests passed fine. I suspect it did not repro because there was not concurrent modification, if we simply pause before iterating to close sessions, it may not do it.

File: http-server/src/main/java/org/triplea/server/http/web/socket/MessageBroadcaster.java [hunk 1]
```diff
@@ -20,7 +21,7 @@
   public void accept(
       final Collection<Session> sessions, final ServerMessageEnvelope serverEventEnvelope) {
     log.info("Broadcasting: {}", serverEventEnvelope);
-    sessions
+    ImmutableSet.copyOf(sessions)
```
[the remark this answers is not in the document]
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/server/http/web/socket/MessageBroadcaster.java]: Why do you suggest that? The unmodifiable variant is intended.

**Comment by DanVanAtta** [comment]
@RoiEXLab we've hit a deadlock in PRs.

https://github.com/triplea-game/triplea/pull/5786 requires a as-of-yet unknown fix.

https://github.com/triplea-game/triplea/pull/5801 requires this PR: https://github.com/triplea-game/triplea/pull/5793 

https://github.com/triplea-game/triplea/pull/5793 seemingly requires: https://github.com/triplea-game/triplea/pull/5802

https://github.com/triplea-game/triplea/pull/5802 would conflict badly with #5786

To be direct:
A) we need to figure out how we can review earlier in coding process so we can give feedback on design
B) when we are reviewing, it's difficult as we mostly gain benefit of knowledge sharing:
  - we review late in the coding process, design changes at that stage are not welcome, it's better to merge and refactor.
  - the items that can be reviewed are going to be pretty minor, line by line changes for sugary syntax is really just not valuable. Our time is spent, trade-off cost to not fix something else, is immensely more valuable

I suspect our CR benefits/goals are really:
A) knowledge sharing
B) pointing out places where code is confusing and asking for it to be improved to be made more clear
C) pointing out wrong code
D) pointing out technical debt addition that can reasonably be fixed

File: http-server/src/main/java/org/triplea/server/http/web/socket/MessageBroadcaster.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (RoiEXLab) [reply http-server/src/main/java/org/triplea/server/http/web/socket/MessageBroadcaster.java]: copyOf does create an unmodifiable copy

**Pull request merged**